nncf/common/tensor_statistics/aggregator.py

```diff
@@ -154,3 +205,44 @@ def _process_outputs(outputs: Any) -> Dict[str, NNCFTensor]:
         :param outputs: raw model outputs
         :return: processed model outputs in Dict[str, Tensor] format
         """
+
+    @abstractmethod
+    def _get_statistics_key(self, statistics: TensorStatistic, target_point: TargetPoint) -> str:
+        """
+        Returns key of statistics.
+
+        :param statistics: Statistics value.
+        :param target_point: Statistics target point.
+        :return: Statistics key.
+        """
+
+
+class StatisticsSerializer:
+    @staticmethod
+    def load_from_file(file_name: str) -> Any:
+        """
+        Loads statistics from a gzip-compressed file.
+        :param file_name: The name of the file from which to load the statistics.
+        :return: The loaded statistics.
+        """
+        try:
+            with gzip.open(file_name, "rb") as f:
+                return pickle.load(f)
+        except FileNotFoundError:
+            raise nncf.ValidationError(f"File not found: {file_name}")
+        except (pickle.UnpicklingError, IOError):
+            raise nncf.ValidationError(f"Error loading statistics from {file_name}")
+
+    @staticmethod
+    def dump_to_file(statistics: Dict[str, TensorType], file_name: str) -> None:
+        """
+        Dumps statistics to a gzip-compressed file.
+        :param data: The statistics to be dumped.
+        :param file_name: The name of the file where the statistics will be dumped.
+        """
+        try:
+            with gzip.open(file_name, "wb") as f:
```

Comment:
Could you consider the approach with dumping each statistic into a separate file which can be interpreted outside of nncf? Some details:
- The statistics path is path to folder
- Each statistic is saved to separate file.
- Introduce flag to dump statistics to file in zipped `numpy.savez` or txt `numpy.savetxt` format. By default zipped format.

cc' @AlexKoff88

Reply:
I agree that more granular access would be helpful from the investigation and memory consumption standpoints. It should be relatively easy to implement as long as we come up with a convenient naming convention for the statistic files. I've heard that OpenVINO has one already so we could use this experience.

Reply:
> The statistics path is path to folder

done

> Each statistic is saved to separate file.

done